src/adapter/src/catalog/storage.rs

```diff
@@ -2002,9 +2116,15 @@ impl<'a> Transaction<'a> {
         schema: &Schema,
     ) -> Result<(), Error> {
         let n = self.schemas.update(|k, _v| {
-            if k.id == schema_id.0 {
+            let (db_id, db_ns) = match database_id {
+                None => (None, None),
```

Comment:
If `database_id` is `None` it shouldn't be a no-op, we'll set `None` for both `database_id` and `database_ns` in `SchemaValue`. Which I think is what we want since we have no info on what database this schema is in?

Reply:
Ah got it, I was misreading this. I thought we were setting the whole `SchemaValue` to `None`.